wdl-engine/src/inputs.rs

```diff
@@ -607,26 +696,46 @@ impl Inputs {
     /// Returns `Ok(None)` if the file contains an empty input.
     pub fn parse(document: &Document, path: impl AsRef<Path>) -> Result<Option<(String, Self)>> {
         let path = path.as_ref();
-        let file = File::open(path).with_context(|| {
-            format!("failed to open input file `{path}`", path = path.display())
-        })?;
+        
+        // Determine the format based on file extension
+        let object = match path.extension().and_then(|ext| ext.to_str()) {
+            Some("json") => JsonInputFormat::parse_file(path)?,
+            Some("yml") | Some("yaml") => YamlInputFormat::parse_file(path)?,
```

Comment:
I think these can just be `Self::parse_json` and `Self::parse_yaml` once we remove `JsonInputFormat` and `YamlInputFormat` and move their implementations into `Self::parse_json` and `Self::parse_yaml`, respectively.